doc/api/readline.md

```diff
@@ -258,6 +258,36 @@ rl.question('What is your favorite food? ', (answer) => {
 });
 ```
 
+The `callback` function passed to `rl.question()` does not follow the typical
+pattern of accepting an `Error` object or `null` as the first argument.
+The `callback` is called with the provided answer as the only argument. If
+the question was canceled by `rl.questionCancel()` the callback will be called
+without arguments.
+
+### `rl.questionCancel()`
+<!-- YAML
+added: REPLACEME
+-->
+
+The `rl.questionCancel()` abort the last call to `rl.question()` if it has
+not already been answered. In that case it does nothing.
+The cancelation will restore the `prompt` and call the `callback` given to
+`rl.question()` without any arguments.
+
+Example usage:
+
+```js
+rl.question('What is your favorite food? ', (answer) => {
+  if (typeof answer === 'undefined') {
+    console.log('The food question was canceled');
+  } else {
+    console.log(`Oh, so your favorite food is ${answer}`);
+  }
+});
+
+rl.questionCancel();
```

Comment:
Nit: could we add a more usable example, i.e. timeout to cancel the question and log that 'question-timeouted'?

Reply:
I will change this to a timeout example as you suggest.